From d2cd228edcee91dea4da21a1c01b6a0fc9be6aae Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Fri, 20 Mar 2015 15:05:55 +0800
Subject: [PATCH] MLK-10450: dma: imx-sdma: enable Mega/Fast support for
 i.mx7d

Enable Mega/Fast support for i.mx7d. Need save and restore SDMA registers.

Signed-off-by: Robin Gong <b38343@freescale.com>
---
 drivers/dma/imx-sdma.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 8a34ad8..526d679 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -2091,8 +2091,8 @@ static int sdma_suspend(struct device *dev)
 	struct sdma_engine *sdma = platform_get_drvdata(pdev);
 	int i;
 
-	/* Do nothing if not i.MX6SX */
-	if (sdma->drvdata != &sdma_imx6sx)
+	/* Do nothing if not i.MX6SX or i.MX7D*/
+	if (sdma->drvdata != &sdma_imx6sx && sdma->drvdata != &sdma_imx7d)
 		return 0;
 
 	clk_enable(sdma->clk_ipg);
@@ -2123,8 +2123,8 @@ static int sdma_resume(struct device *dev)
 	struct sdma_engine *sdma = platform_get_drvdata(pdev);
 	int i, ret;
 
-	/* Do nothing if not i.MX6SX */
-	if (sdma->drvdata != &sdma_imx6sx)
+	/* Do nothing if not i.MX6SX or i.MX7D*/
+	if (sdma->drvdata != &sdma_imx6sx && sdma->drvdata != &sdma_imx7d)
 		return 0;
 
 	clk_enable(sdma->clk_ipg);
-- 
1.8.0

